task8-rashid-ab/src/main.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Producer.h"
+#include <iostream>
+#include <vector>
+#include <future>
+#include <algorithm>
+
+/**
+ * The main function of the program.
+ *
+ * It creates producer threads to generate values using the Producer class,
+ * and a consumer thread that remains detached.
+ *
+ * @return 0 on successful execution.
+ */
+int main() {
+    std::vector<double> values = {28312, 38348, 99341};
+    std::vector<std::future<void>> producerFutures;      // Store futures of producer threads
+    std::thread consumer([]() {});
```

Comment:
The consumer was supposed to get the messages and print them. It definitely wasn't supposed to do nothing..